src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,113 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_FACTOR = 2;
+    private Entry<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public MyHashMap(int initialCapacity) {
+        buckets = new Entry[initialCapacity];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Entry<K, V> entry = buckets[index];
+        while (entry != null) {
+            if (key == entry.key || key != null && key.equals(entry.key)) {
+                entry.value = value;
+                return;
+            }
+            entry = entry.next;
+        }
+        Entry<K,V> newEntry = new Entry<>(key, value);
+        newEntry.next = buckets[index];
+        buckets[index] = newEntry;
+        size++;
+        if ((double) size / buckets.length > LOAD_FACTOR) {
+            resize();
+        }
```

Comment:
do this (check and resize if required) at the start of this method
```suggestion
        if (getThreshold() > LOAD_FACTOR) {
            resize();
        }
```